### PR TITLE
doc(development guide): Add information about form fields in dev guide

### PR DESCRIPTION
**Pull request by fivetran-satvikpatil**
Ticket: T-797928

### Changes
Add information about form fields in the development guide, and another few doc changes

### CODE REVIEW

**Comment by fivetran-satvikpatil** [comment]
@fivetran-dejantucakov Please review this PR.

**Comment by fivetran-dejantucakov** [comment]
@5tran-alexil Can you please take this one?
Cc: @fivetran-satvikpatil

File: development-guide.md [hunk 1]
```diff
@@ -80,14 +80,37 @@ The following are hard requirements to be able to deploy partner code to Fivetra
...
+- Use clear and descriptive labels for each form field. Avoid technical jargon if possible.
+- Organize the fields in a logical order that reflects the setup process.
 
-## Connector guidelines
+### RPC Calls
+#### ConfigurationForm
+This operation retrieves all the setup form fields and tests information. You can provide various parameters for the fields to enhance the user experience, such as descriptions, optional fields, and more.
+
+#### Test
+The previous RPC call retrieves the tests that need to be executed during connection setup. This operation then invokes the test with the customer's credentials as parameters. Finally, it should return a success or failure indication for the test execution.
```
Comment (5tran-alexil) [inline development-guide.md:93]:
```suggestion
The [`ConfigurationForm` RPC call](#configurationform) retrieves the tests that need to be executed during connection setup. The `Test` call then invokes the test with the customer's credentials as parameters. As a result, it should return a success or failure indication for the test execution.
```

File: development-guide.md [hunk 2]
```diff
@@ -80,14 +80,37 @@ The following are hard requirements to be able to deploy partner code to Fivetra
...
+
+### Supported setup form fields 
+- Text Field: A standard text input field for user text entry.
+- Dropdown: A dropdown selection menu that allows users to choose one option from the list provided by you.
```
Comment (5tran-alexil) [inline development-guide.md:97]:
```suggestion
- Dropdown: A drop-down menu that allows users to choose one option from the list provided by you.
```

File: development-guide.md [hunk 3]
```diff
@@ -80,14 +80,37 @@ The following are hard requirements to be able to deploy partner code to Fivetra
...
+## Source Connector guidelines
 
 - Don't push anything other than source data to the destination. State data is saved to production database and returned in `UpdateRequest`.
 - Don't forget to handle new schemas/tables/columns per the information and user choices in `UpdateRequest#selection`.
 - Make sure you checkpoint at least once an hour. In general, the more frequently you do it, the better.
 
-## Destination guidelines
+### RPC Calls
+#### Schema
+This operation retrieves the customer's schemas, tables, and columns. It also includes an optional `selection_not_supported` field that indicates whether customers can select or deselect tables and columns within the Fivetran dashboard.
```
Comment (5tran-alexil) [inline development-guide.md:108]:
```suggestion
The `Schema` RPC call retrieves the user's schemas, tables, and columns. It also includes an optional `selection_not_supported` field that indicates whether the user can select or deselect tables and columns within the Fivetran dashboard.
```

File: development-guide.md [hunk 4]
```diff
@@ -80,14 +80,37 @@ The following are hard requirements to be able to deploy partner code to Fivetra
...
+
+#### Update
+This operation should retrieve data from the source. We send a request using the `UpdateRequest` message, which includes the customer's state, credentials, and schema information. The response, streaming through the `UpdateResponse` message, can contain data records and other supported operations.
```
Comment (5tran-alexil) [inline development-guide.md:111]:
```suggestion
The `Update` RPC call should retrieve data from the source. We send a request using the `UpdateRequest` message, which includes the user's connection state, credentials, and schema information. The response, streaming through the `UpdateResponse` message, can contain data records and other supported operations.
```

File: development-guide.md [hunk 5]
```diff
@@ -130,12 +153,18 @@ This operation should report all columns in the destination table, including Fiv
 - However, this operation should not drop any columns even if the `AlterTable` request has a table with a different set of columns. Dropping columns could lead to unexpected customer data loss and is against Fivetran's general approach to data movement.
 
 #### WriteBatchRequest
-- `replace_files` is for the `upsert` operation where the rows should be inserted if they don't exist or updated if they do. Each row always provides values for all columns. Populate the `_fivetran_synced` column in the destination with the values coming in from the CSV files.
+This operation provides details about the batch files containing the records to be pushed to the destination. We provide the `WriteBatchRequest` parameter that contains all the information required for you to read the batch files. Here are some of the fields included in the request message:
```
Comment (5tran-alexil) [inline development-guide.md:156]:
```suggestion
The `WriteBatchRequest` RPC call provides details about the batch files containing the records to be pushed to the destination. We provide the `WriteBatchRequest` parameter that contains all the information required for you to read the batch files. Here are some of the fields included in the request message:
```

Comment (5tran-alexil) [reply development-guide.md:156]: @fivetran-satvikpatil We might use 'The `CallName` RPC call ..." wording across the document.

**Review by 5tran-alexil — COMMENTED** [review]
@fivetran-satvikpatil  see my comments inline, looks good.

File: development-guide.md [hunk 6]
```diff
@@ -80,14 +80,37 @@ The following are hard requirements to be able to deploy partner code to Fivetra
...
-## Connector guidelines
+### RPC Calls
+#### ConfigurationForm
+This operation retrieves all the setup form fields and tests information. You can provide various parameters for the fields to enhance the user experience, such as descriptions, optional fields, and more.
+
+#### Test
+The previous RPC call retrieves the tests that need to be executed during connection setup. This operation then invokes the test with the customer's credentials as parameters. Finally, it should return a success or failure indication for the test execution.
+
+### Supported setup form fields 
+- Text Field: A standard text input field for user text entry.
```
Comment (alsnkline) [inline development-guide.md:96]: Can we add the Text fields available for this:
You provide a 'title' displayed above the field and can optionally provide a 'description' upto x characters long which is displayed below the field and can help explain what should be completed.

**Review by alsnkline — APPROVED** [review]
Great job. one minor suggestion

File: development-guide.md (hunk 6 above)
Comment (fivetran-satvikpatil) [reply development-guide.md]: Added

**Review by varundhall — APPROVED** [review]

File: development-guide.md [hunk 7]
```diff
@@ -80,14 +80,37 @@ The following are hard requirements to be able to deploy partner code to Fivetra
...
-## Connector guidelines
+### RPC Calls
+#### ConfigurationForm
+The `ConfigurationForm` RPC call retrieves all the setup form fields and tests information. You can provide various parameters for the fields to enhance the user experience, such as descriptions, optional fields, and more.
+
+#### Test
+The [`ConfigurationForm` RPC call](#configurationform) retrieves the tests that need to be executed during connection setup. The `Test` call then invokes the test with the customer's credentials as parameters. As a result, it should return a success or failure indication for the test execution.
+
+### Supported setup form fields 
+g- Text Field: A standard text input field for user text entry. You can provide a `title` displayed above the field. You can indicate whether the field is `required`, and you may also include an optional `description` displayed below the field to help explain what should be completed by the user.
```
Comment (5tran-alexil) [inline development-guide.md:96]: @fivetran-satvikpatil Is this a typo?

Comment (fivetran-satvikpatil) [reply development-guide.md:96]: Yes it is a typo

Comment (fivetran-satvikpatil) [reply development-guide.md]: Updated it.

**Review by 5tran-alexil — APPROVED** [review]
LGTM

**Pull request merged**